service/src/main/java/org/broadinstitute/listener/relay/inspectors/TokenChecker.java

```diff
@@ -16,43 +14,58 @@
 
 @Component
 public class TokenChecker {
-  private final HttpClient httpClient = HttpClient.newHttpClient();
-  private final String GOOGLE_OAUTH_SERVER =
-      "https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=";
   private final Logger logger = LoggerFactory.getLogger(SamResourceClient.class);
+  private final GoogleTokenInfoClient googleTokenInfoClient;
 
-  public OauthInfo getOauthInfo(String token) throws IOException, InterruptedException {
-    var jwtExpiration = tryDecodeAsB2cToken(token);
+  public TokenChecker(GoogleTokenInfoClient googleTokenInfoClient) {
```

Comment:
I extracted the "get token info from the google APIs" bits to a separate class to ease testability.